server/src/main/java/org/elasticsearch/search/aggregations/AggregationPhase.java

```diff
@@ -37,10 +38,21 @@ public void preProcess(SearchContext context) {
         } catch (IOException e) {
             throw new AggregationInitializationException("Could not initialize aggregators", e);
         }
-        Collector collector = context.getProfilers() == null
-            ? bucketCollector
-            : new InternalProfileCollector(bucketCollector, CollectorResult.REASON_AGGREGATION, List.of());
-        context.queryCollectors().put(AggregationPhase.class, collector);
+        if (context.aggregations().factories().context() != null
+            && context.aggregations().factories().context().isInOrderExecutionRequired()) {
+            TimeSeriesIndexSearcher searcher = new TimeSeriesIndexSearcher(context.searcher());
+            try {
+                searcher.search(context.rewrittenQuery(), bucketCollector);
+            } catch (IOException e) {
+                throw new AggregationExecutionException("Could not perform time series aggregation", e);
+            }
+            context.queryCollectors().put(AggregationPhase.class, BucketCollector.NO_OP_COLLECTOR);
+        } else {
+            Collector collector = context.getProfilers() == null
+                ? bucketCollector
+                : new InternalProfileCollector(bucketCollector, CollectorResult.REASON_AGGREGATION, List.of());
```

Comment:
I think this should be outside the `if` statement - we want it even if we're collecting in order.

Reply:
Ah. I see, you don't have the right sub-type. Well, maybe a follow up?

Reply:
Yeah, will deal with profilers later.